### PR TITLE
LocalStore-only queries

### PR DESCRIPTION
**Pull request by schmidt-sebastian**
This removes FirestoreClient from the firestore-exp build. FirestoreClient depends on all components in the system, and as such it is not very well geared for a tree-shakeable SDK.

Instead, the firestore-exp SDK now purely relies on Component Providers and splits up offline-only and online use. This allows LocalStore-only queries in getDocFromCache and getDocsFromCache, dropping their size from 220Kb to 130Kb (ish).

Couple things:
- I moved the component configuration into the main Firestore class, which has some negative size consequences (pulls in AutoId and other things). On the whole, using the SDK without AutoId (needed for SyncEngine/Persistence/`doc()`) is pretty impossible though and hence I don't think this affects users bundle sizes.
- I fixed a bug in scripts/build-bundle.ts, which meant I had to regen the scripts/*.js files.

### CODE REVIEW

File: packages/firestore/lite/src/api/database.ts [hunk 1]
```diff
@@ -111,7 +111,6 @@ export class Firestore
    * Only ever called once.
    */
   protected _terminate(): Promise<void> {
-    debugAssert(!this._terminated, 'Cannot invoke _terminate() more than once');
```
Comment (schmidt-sebastian) [inline packages/firestore/lite/src/api/database.ts:114]: This doesn't work anymore as this is not called from the AsyncQueue (by the overridden method)

File: packages/firestore/exp/src/api/components.ts [hunk 2]
```diff
@@ -16,138 +16,183 @@
...
-  const firestoreClientPromise = firestoreClientInstances.get(firestore);
-  if (firestoreClientPromise) {
-    logDebug(LOG_TAG, 'Removing FirestoreClient');
-    firestoreClientInstances.delete(firestore);
-    return (await firestoreClientPromise).terminate();
+  const onlineComponentProviderPromise = onlineComponentProviders.get(
+    firestore
+  );
+  if (onlineComponentProviderPromise) {
+    logDebug(LOG_TAG, 'Removing OnlineComponentProvider');
```
Comment (var-const) [inline packages/firestore/exp/src/api/components.ts:185]: Question: nothing is logged if the promise was not found -- is this intentional?

Comment (schmidt-sebastian) [reply packages/firestore/exp/src/api/components.ts:185]: Yeah, I think that is ok. We only log when components are initialized and when they are terminated.

File: packages/firestore/exp/src/api/database.ts [hunk 3]
```diff
@@ -108,41 +172,55 @@ export function enableIndexedDbPersistence(
...
 }
 
 export function enableMultiTabIndexedDbPersistence(
   firestore: firestore.FirebaseFirestore
 ): Promise<void> {
   const firestoreImpl = cast(firestore, Firestore);
+  verifyNotInitialized(firestoreImpl);
+  const settings = firestoreImpl._getSettings();
+
+  // `_getSettings()` freezes the client settings and prevents further changes
```
[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply packages/firestore/exp/src/api/database.ts:203]: Updated.

File: packages/firestore/exp/src/api/database.ts [hunk 4]
```diff
@@ -179,26 +257,42 @@ export function waitForPendingWrites(
...
-    firestoreClient.enableNetwork()
+  return Promise.all([
+    getRemoteStore(firestoreImpl),
+    getPersistence(firestoreImpl)
+  ]).then(([remoteStore, persistence]) =>
+    enqueueNetworkEnabled(
+      firestoreImpl._queue,
+      remoteStore,
+      persistence,
+      /* enabled=*/ true
```
[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply packages/firestore/exp/src/api/database.ts]: :) Fixed here and below.

File: packages/firestore/exp/src/api/reference.ts [hunk 5]
```diff
@@ -54,57 +54,67 @@ import {
...
 export function getDoc<T>(
   reference: firestore.DocumentReference<T>
 ): Promise<firestore.DocumentSnapshot<T>> {
   const ref = cast<DocumentReference<T>>(reference, DocumentReference);
   const firestore = cast<Firestore>(ref.firestore, Firestore);
-  return getFirestoreClient(firestore).then(async firestoreClient => {
-    const viewSnapshot = await firestoreClient.getDocumentFromLocalCache(
-      firestoreClient,
+  return getEventManager(firestore).then(eventManager =>
+    enqueueReadDocumentViaSnapshotListener(
```
[the remark this answers is not in the document]
Comment (schmidt-sebastian) [reply packages/firestore/exp/src/api/reference.ts:75]: This is `getDoc`, which does a "default" read (from Server first but with the option of reading from cache if offline). The one below is `getDocFromCache`, which only reads from cache.

**Pull request merged**